### PR TITLE
[v2.9] Added proxied rancher provisioning tests

### PR DESCRIPTION
**Pull request by MSpencer87**
## Issue: 

Added proxied rancher provisioning tests for https://github.com/rancher/qa-tasks/issues/479

Tests added:
- rke1 node driver
- rke1 custom clusters
- rke2 node driver
- rke2 custom clusters
- k3s node driver
- k3s custom cluster
 
## QA Testing Considerations

 Runs in Jenkins

### CODE REVIEW

File: go.mod
Comment (markusewalker) [inline go.mod]: This should be able to be removed.

File: go.mod
Comment (markusewalker) [inline go.mod:18]: This comment needs to be removed.

File: go.mod
Comment (markusewalker) [inline go.mod]: Just a reminder that once the shepherd PR is merged, this will need to point to the updated `main` branch commit and not your forked repo.

File: tests/v2/validation/provisioning/proxy/k3s_proxy_node_driver_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/proxy/k3s_proxy_node_driver_test.go:84]: Can you change this to be the standard user and not the admin user?

File: tests/v2/validation/provisioning/proxy/rke1_proxy_custom_cluster_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/proxy/rke1_proxy_custom_cluster_test.go:95]: Same note with the admin user being standard instead.

File: tests/v2/validation/provisioning/proxy/rke1_proxy_node_driver_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/proxy/rke1_proxy_node_driver_test.go:85]: Same note with the admin user being standard instead.

File: tests/v2/validation/provisioning/proxy/rke2_proxy_custom_cluster_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/proxy/rke2_proxy_custom_cluster_test.go:102]: Same note with the admin user being standard instead.

File: tests/v2/validation/provisioning/proxy/rke2_proxy_node_driver_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/proxy/rke2_proxy_node_driver_test.go:84]: Same note with the admin user being standard instead.

File: tests/v2/validation/provisioning/proxy/k3s_proxy_node_driver_test.go
Comment (MSpencer87) [reply tests/v2/validation/provisioning/proxy/k3s_proxy_node_driver_test.go]: Updated for standard user

File: tests/v2/validation/provisioning/proxy/rke1_proxy_node_driver_test.go
Comment (MSpencer87) [reply tests/v2/validation/provisioning/proxy/rke1_proxy_node_driver_test.go]: Updated for standard user

File: tests/v2/validation/provisioning/proxy/rke2_proxy_node_driver_test.go
Comment (MSpencer87) [reply tests/v2/validation/provisioning/proxy/rke2_proxy_node_driver_test.go]: Updated for standard user

File: tests/v2/validation/provisioning/proxy/rke2_proxy_custom_cluster_test.go
Comment (MSpencer87) [reply tests/v2/validation/provisioning/proxy/rke2_proxy_custom_cluster_test.go]: Updated for standard user

File: tests/v2/validation/provisioning/proxy/rke1_proxy_custom_cluster_test.go
Comment (MSpencer87) [reply tests/v2/validation/provisioning/proxy/rke1_proxy_custom_cluster_test.go]: Updated for standard user

**Review by markusewalker — COMMENTED** [review]
LGTM, will approve once rebased.

**Review by markusewalker — APPROVED** [review]
LGTM

**Review by Josh-Diamond — APPROVED** [review]

**Pull request merged**
